### PR TITLE
15.0 fix project forecast perf issues

### PR DESCRIPTION
**Pull request by gurneyalex**

### CODE REVIEW

**Comment by yankinmax** [comment]
@gurneyalex overall LGTM. Just a question, because I was solving unlink issue here:
https://github.com/OCA/project/pull/1044
Does `ondelete=cascade` resolve issue with the removal of forecast lines for non-existing related records?

File: project_forecast_line/models/forecast_line.py [hunk 1]
```diff
@@ -202,7 +224,51 @@ def _compute_consolidated_forecast(self):
...
+        for vals in values:
+            start_date = vals["date_from"]
+            rec = self_by_start_date.pop(start_date, None)
+            if rec is None:
+                to_create.append(vals)
+            else:
+                rec.write(vals)
+                updated.append(rec.id)
+        _logger.debug("updated lines %s", updated)
+        to_remove = self.browse([r.id for r in self_by_start_date.values()])
```
Comment (damdam-s) [inline project_forecast_line/models/forecast_line.py:266]:
```suggestion
        to_remove = self.browse([r.id for r in self_by_start_date.values()])
        _logger.debug("removed lines %s", to_remove.ids)
```

**Review by damdam-s — APPROVED** [review]
LG

File: project_forecast_line/models/forecast_line_mixin.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    def _get_forecast_lines(self, domain=None):
+        self.ensure_one()
+        base_domain = [("res.model", "=", self._name), ("res_id", "=", self.id)]
+        if domain is not None:
+            base_domain += domain
+        return self.env["forecast.line"].search(base_domain)
+
+    @api.model
+    def _recompute_forecast_lines(self, force_company_id=None):
+        return super()._recompute_forecast_lines(force_company_id=force_company_id)
```
Comment (jcoux) [inline project_forecast_line/models/forecast_line_mixin.py]: Seems you did nothing more than call the super with the same arguments, maybe you can remove this?
```suggestion
```

File: project_forecast_line/models/forecast_line_mixin.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    def _update_forecast_lines(self, **kwargs):
+        return super()._update_forecast_lines(**kwargs)
```
Comment (jcoux) [inline project_forecast_line/models/forecast_line_mixin.py]: Seems you did nothing more than call the super with the same arguments, maybe you can remove this?
```suggestion
```

**Review by jcoux — APPROVED** [review]
Technically seems good :+1:

File: project_forecast_line/models/forecast_line.py (hunk 1 above)
Comment (yankinmax) [reply project_forecast_line/models/forecast_line.py:266]: updated

File: project_forecast_line/models/forecast_line_mixin.py (hunk 2 above)
Comment (yankinmax) [reply project_forecast_line/models/forecast_line_mixin.py]: updated

File: project_forecast_line/models/forecast_line_mixin.py (hunk 3 above)
Comment (yankinmax) [reply project_forecast_line/models/forecast_line_mixin.py]: updated

**Review by yankinmax — APPROVED** [review]

**Pull request merged**